public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/KeyValuesTable.tsx

```diff
@@ -17,6 +17,8 @@ import cx from 'classnames';
 import * as React from 'react';
 
 import { GrafanaTheme2 } from '@grafana/data';
+import { FlameGraph } from '@grafana/flamegraph';
```

Comment:
Is it intended to have a direct dependency o traceview->flamegraph?  Do you think `<PanelRenderer  pluginId="flamegraph" />` would work here? Also, the `FlameGraphPanel` has some error/field checking that is not included if raw FlameGraph is used.

Reply:
So I spent some time trying to use the `<PanelRenderer />` but honestly it's kind of tricky. It requires a width and height first of all which means we need to calculate the size of the flame graph based on depth and anything else we render around the flame graph. This seems prone to error and miscalculation. Second for every span row we calculate height based on span details open/closed and the height of the span detail content and re-render list-view if content toggle in span details is clicked for example as this shifts elements y positions. With `<PanelRenderer />` it's tricky because the panel is loaded asynchronously so it's not present in the DOM when the list view is calculating heights.

I think for error checking we are ok because `checkFields` is also done in `FlameGraphDataContainer` used by `<FlameGraph />` so will error if fields incorrect.